### PR TITLE
Add Basic Rust checks

### PR DESCRIPTION
**Pull request by mythi**
Fixes: #71
Fixes: #105

### CODE REVIEW

File: src/enclave-agent/Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,10 @@ tokio = { version = "1.14.0", features = ["rt-multi-thread", "macros", "signal"]
 
 ttrpc = { git = "https://github.com/haosanzi/ttrpc-rust", features = ["async"] }
 
+[features]
+default = ["image-rs/enclave-cc", "image-rs/signature-cosign"]
+simple = ["image-rs/snapshot-unionfs", "image-rs/signature-simple", "image-rs/keywrap-native", "image-rs/encryption"]
```
[the remark this answers is not in the document]
Comment (mythi) [reply src/enclave-agent/Cargo.toml:30]: the idea of `simple` was just to be able to build with features that do not pull in RATS-TLS. I did not try adding `signature-cosign` to `simple` features...

[the remark this answers is not in the document]
Comment (mythi) [reply src/enclave-agent/Cargo.toml:30]: I remember some comment from long time back that `signature-cosign` would pull in `eaa_kbc`. Is that not true (anymore)?

[the remark this answers is not in the document]
Comment (Xynnn007) [reply src/enclave-agent/Cargo.toml:30]: Oh, I see. In fact `eaa_kbc` is pulled by `attestation-agent/eaa-kbc` here https://github.com/confidential-containers/image-rs/blob/main/Cargo.toml#L77 . `signature-cosign` and `signature-simple` are only image signing schemes, not related to the remote attestation : )

[the remark this answers is not in the document]
Comment (mythi) [reply src/enclave-agent/Cargo.toml]: I added `image-rs/signature-cosign` and it builds OK.

**Review by haosanzi — APPROVED** [review]

**Review by dcmiddle — APPROVED** [review]

**Pull request merged**